### PR TITLE
Front: scenario explorator / correct STDCM

### PR DESCRIPTION
**Pull request by nicolaswurtz**

### CODE REVIEW

File: front/src/applications/operationalStudies/views/Project.js [hunk 1]
```diff
@@ -60,10 +61,18 @@ export default function Project() {
     },
   ];
 
+  const getProjectImage = async (url) => {
+    const image = await get(url, { responseType: 'blob' });
+    setImageUrl(URL.createObjectURL(image));
```
Comment (alexandredamiron) [inline front/src/applications/operationalStudies/views/Project.js]: please catch an eventual problem. I had a crash

File: front/public/locales/fr/common/scenarioExplorator.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  "infraLegend": "infra",
+  "noScenarioSelected": "Cliquez ici pour sélectionner un scénario afin de déterminer une infrastructure et une grille horaire.",
+  "projectLegend": "projet",
+  "projects": "Projets",
+  "scenarioExplorator": "Navigateur de scénarios",
+  "scenarioLegend": "scénario",
+  "scenarios": "Scénarios",
+  "studies": "Études",
+  "studyLegend": "étude"
+}
```
Comment (clarani) [inline front/public/locales/fr/common/scenarioExplorator.json]: missing new line

File: front/src/applications/operationalStudies/components/Project/StudyCard.tsx [hunk 3]
```diff
@@ -55,7 +56,7 @@ export default function StudyCard({ setFilterChips, study }: Props) {
         </button>
       </div>
       <div className="studies-list-card-type">
-        {t(`operationalStudies/study:studyCategories.${study.type}`)}
+        {study.type ? t(`operationalStudies/study:studyCategories.${study.type}`) : ' '}
```
Comment (clarani) [inline front/src/applications/operationalStudies/components/Project/StudyCard.tsx]:
```suggestion
        {study.type ? t(`operationalStudies/study:studyCategories.${study.type}`) : ''}
```

File: front/src/common/ScenarioExplorator/ScenarioExplorator.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      const result = await get(url);
+      setDetails(result);
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  const getProjectImage = async (url: string) => {
+    const image = await get(url, { responseType: 'blob' });
+    setImageUrl(URL.createObjectURL(image));
```
Comment (clarani) [inline front/src/common/ScenarioExplorator/ScenarioExplorator.tsx]: Can add a try/catch here too

File: front/src/common/ScenarioExplorator/ScenarioExplorator.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  const { openModal } = useContext(ModalContext);
+  const projectID = useSelector(getProjectID);
+  const studyID = useSelector(getStudyID);
+  const scenarioID = useSelector(getScenarioID);
+  const [projectDetails, setProjectDetails] = useState<any>();
+  const [studyDetails, setStudyDetails] = useState<any>();
+  const [scenarioDetails, setScenarioDetails] = useState<any>();
+  const [imageUrl, setImageUrl] = useState<string>();
+
+  async function getProjectStudyScenarioDetail(url: string, setDetails: (arg0: any) => void) {
```
Comment (clarani) [inline front/src/common/ScenarioExplorator/ScenarioExplorator.tsx]: The name of this function is very confusing. I think getDetails would be more appropriate

File: front/src/common/ScenarioExplorator/ScenarioExploratorModal.tsx [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+                    {projectsList.length}
+                  </span>
+                )}
+              </div>
+              <div className="scenario-explorator-modal-part-itemslist">
+                {useMemo(
+                  () =>
+                    projectsList &&
+                    projectsList.map((project) => (
+                      <ProjectMiniCard
```
Comment (clarani) [inline front/src/common/ScenarioExplorator/ScenarioExploratorModal.tsx:112]: Should memo this ProjectMiniCard too if possible

File: front/src/common/ScenarioExplorator/ScenarioExploratorModal.tsx [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+                    {studiesList.length}
+                  </span>
+                )}
+              </div>
+              <div className="scenario-explorator-modal-part-itemslist">
+                {useMemo(
+                  () =>
+                    studiesList &&
+                    studiesList.map((study) => (
+                      <StudyMiniCard
```
Comment (clarani) [inline front/src/common/ScenarioExplorator/ScenarioExploratorModal.tsx:143]: Same here for StudyMiniCard

File: front/src/common/ScenarioExplorator/ScenarioExploratorModal.tsx [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+                )}
+              </div>
+              <div className="scenario-explorator-modal-part-itemslist">
+                {useMemo(
+                  () =>
+                    projectID &&
+                    studyID &&
+                    scenariosList &&
+                    scenariosList.map((scenario) => (
+                      <ScenarioMiniCard
```
Comment (clarani) [inline front/src/common/ScenarioExplorator/ScenarioExploratorModal.tsx:176]: Same here

File: front/src/common/ScenarioExplorator/ScenarioExploratorProject2Image.tsx [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+import { PROJECTS_URI } from 'applications/operationalStudies/components/operationalStudiesConsts';
+import { get } from 'common/requests';
+import { MiniCardsImageProps } from './ScenarioExploratorTypes';
+
+function Project2Image({ project }: MiniCardsImageProps) {
+  const [imageUrl, setImageUrl] = useState<string>();
+
+  useMemo(async () => {
+    if (!project || !project.image_url) return;
+    const image = await get(`${PROJECTS_URI}${project.id}/image/`, { responseType: 'blob' });
```
Comment (clarani) [inline front/src/common/ScenarioExplorator/ScenarioExploratorProject2Image.tsx]: Add try/catch here

File: front/src/common/ScenarioExplorator/ScenarioExploratorProject2Image.tsx [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import React, { useMemo, useState } from 'react';
+import { PROJECTS_URI } from 'applications/operationalStudies/components/operationalStudiesConsts';
+import { get } from 'common/requests';
+import { MiniCardsImageProps } from './ScenarioExploratorTypes';
+
+function Project2Image({ project }: MiniCardsImageProps) {
+  const [imageUrl, setImageUrl] = useState<string>();
+
+  useMemo(async () => {
```
Comment (clarani) [inline front/src/common/ScenarioExplorator/ScenarioExploratorProject2Image.tsx:9]: Are you sure useMemo is needed here as the whole component has already been memoized ?

File: front/src/styles/scss/common/components/_scenarioExplorator.scss [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+          }
+        }
+        &.empty {
+          cursor: not-allowed;
+          opacity: 0.5;
+        }
+      }
+    }
+  }
+}
```
Comment (clarani) [inline front/src/styles/scss/common/components/_scenarioExplorator.scss]: missing new line

File: front/src/styles/scss/common/components/_scenarioExplorator.scss [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+  }
+  &:hover {
+    opacity: 0.5;
+  }
+}
+.scenario-explorator-modal {
+  .scenario-explorator-modal-title {
+    display: flex;
+    align-items: center;
+    gap: 1rem;
```
Comment (clarani) [inline front/src/styles/scss/common/components/_scenarioExplorator.scss:111]: Not useful I think, can test without it :) 
```suggestion
```

**Review by clarani — CHANGES_REQUESTED** [review]
Good job ! I left some comments and found 2 small bugs (I think)

StudyTypes.flowRate on the top-right hand corner
image

I'm not sure whether it is the expected behaviour or not, but I have this page (no scenario selected, but the simulation is open :thinking: )
image

File: front/src/applications/operationalStudies/views/Project.js (hunk 1 above)
Comment (nicolaswurtz) [reply front/src/applications/operationalStudies/views/Project.js]: yes !

File: front/public/locales/fr/common/scenarioExplorator.json (hunk 2 above)
Comment (nicolaswurtz) [reply front/public/locales/fr/common/scenarioExplorator.json]: Thx ! All is corrected :)

File: front/src/applications/operationalStudies/components/Project/StudyCard.tsx (hunk 3 above)
Comment (nicolaswurtz) [reply front/src/applications/operationalStudies/components/Project/StudyCard.tsx]: it was a bit of a slog but it's OK I did it differently to keep the size fixed if the div was empty :P

File: front/src/common/ScenarioExplorator/ScenarioExplorator.tsx (hunk 5 above)
Comment (nicolaswurtz) [reply front/src/common/ScenarioExplorator/ScenarioExplorator.tsx]: Agree !

File: front/src/common/ScenarioExplorator/ScenarioExplorator.tsx (hunk 4 above)
Comment (nicolaswurtz) [reply front/src/common/ScenarioExplorator/ScenarioExplorator.tsx]: Done

File: front/src/common/ScenarioExplorator/ScenarioExploratorModal.tsx (hunk 6 above)
Comment (nicolaswurtz) [reply front/src/common/ScenarioExplorator/ScenarioExploratorModal.tsx:112]: Yes but I've tried a lot of solutions, none was efficient, I need to see this with you (I told you before, it was this point !)

File: front/src/common/ScenarioExplorator/ScenarioExploratorModal.tsx (hunk 7 above)
Comment (nicolaswurtz) [reply front/src/common/ScenarioExplorator/ScenarioExploratorModal.tsx:143]: Yes but I've tried a lot of solutions, none was efficient, I need to see this with you (I told you before, it was this point !)

File: front/src/common/ScenarioExplorator/ScenarioExploratorModal.tsx (hunk 8 above)
Comment (nicolaswurtz) [reply front/src/common/ScenarioExplorator/ScenarioExploratorModal.tsx:176]: Yes but I've tried a lot of solutions, none was efficient, I need to see this with you (I told you before, it was this point !)

File: front/src/common/ScenarioExplorator/ScenarioExploratorProject2Image.tsx (hunk 9 above)
Comment (nicolaswurtz) [reply front/src/common/ScenarioExplorator/ScenarioExploratorProject2Image.tsx]: yep

File: front/src/common/ScenarioExplorator/ScenarioExploratorProject2Image.tsx (hunk 10 above)
Comment (nicolaswurtz) [reply front/src/common/ScenarioExplorator/ScenarioExploratorProject2Image.tsx:9]: You're right, some cleaning from my tries was not completely done

File: front/src/styles/scss/common/components/_scenarioExplorator.scss (hunk 12 above)
Comment (nicolaswurtz) [reply front/src/styles/scss/common/components/_scenarioExplorator.scss:111]: He's useful !

File: front/src/styles/scss/common/components/_scenarioExplorator.scss (hunk 11 above)
Comment (nicolaswurtz) [reply front/src/styles/scss/common/components/_scenarioExplorator.scss]: ok

**Review by clarani — APPROVED** [review]

**Review by Tguisnet — APPROVED** [review]

**Review by eckter — APPROVED** [review]

**Pull request merged**